lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -339,80 +269,43 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     } finally {
       perf.latencyInMs = Date.now() - perf.startTimestampInMs
       this.checkLastCallPerformance(perf)
-      this.updateHealthyStatus()
       if (this.enableDbSync) {
-        if (
-          !this.syncingDb &&
-          MAJOR_METHOD_NAMES.includes(perf.methodName) &&
-          this.hasEnoughWaitSinceLastDbSync(1000 * this.config.DB_SYNC_INTERVAL_IN_S)
-        ) {
+        if (!this.syncingDb && this.hasEnoughWaitSinceLastDbSync(1000 * this.config.DB_SYNC_INTERVAL_IN_S)) {
           this.logDbSyncRequested()
           if (Math.random() < this.dbSyncSampleProb) {
             this.logDbSyncSampled()
             this.syncingDb = true
             // Fire and forget. Won't check the sync result.
-            this.syncAndUpdateProviderState()
+            this.syncAndUpdateProviderHealthState()
           }
         }
       }
-      this.lastCallTimestampInMs = perf.startTimestampInMs
     }
   }
 
-  private async syncAndUpdateProviderState() {
+  private async syncAndUpdateProviderHealthState() {
```

Comment:
so every `dbSyncSampleProb` you check to see if the health state has changed? and if so, update it? Do we still need the boolean `this.syncingDb`? Whats the expected logic here?

Reply:
Both are needed and they serve different purposes.

`this.syncingDb` serves as a lock to ensure that only 1 DB read will happen even if the lambda is processing multiple RPC requests. Without it, before the read of first DB read finishes, more DB read will be made. 

`dbSyncSampleProb` is to reduce the spike of huge number of lambda doing DB read at the same time. The use of `this.syncingDb` and its sync interval still cannot prevent all lambdas doing DB read at the same time, which could be a dangerous load to DB.

This is essentially the idea I discussed in the design doc https://www.notion.so/uniswaplabs/Rate-Limit-DB-Sync-in-RPC-Gateway-40fec8f3ab7e49fca8b3471586b55970?pvs=4